### ISSUE TITLE
getPaymentsOperationHistory

### ISSUE THREAD

**Issue by ricarderobl** [allegro/allegro-api#8512]
W jaki sposób mogę połączyć historię operacji z id zamówień? Nie widzę jak mógłbym to zrobić.
Z metody payments/payment-operations mogę pobrać tylko id płatności nic więcej. A chcę płatności łączyć z konkretnymi zamówieniami, bo one są do konkretnych zamówień tj. wpłaty są do konkretnych i zwroty są do konkretnych. Czemu id zamówienia nie widnieje przy tej metodzie?

Ale przede wszystkim czy da się innymi metodami połączyć ID płatności z ID zamówienia?

**Comment by Lukasz-Zurek** [comment]
W historii operacji płatniczych nie zwracamy identyfikatorów zamówień. Zwracamy jednak inne dane np. identyfikator płatności (payment.id) czy też login kupującego i po tych właśnie danych możesz wyfiltrować listę zamówień dostępną w ramach [GET /order/checkout-forms](https://developer.allegro.pl/documentation/#operation/getListOfOrdersUsingGET).

**Comment by Lukasz-Zurek** [comment]
> Nadal nie rozumiem, bo pytanie jest szersze niż tylko płatności za dane zamówienie chodzi też o zwroty.
> 
> Przykład. Kupujący Client:111111111 zamówił towar dnia 2024-02-01 na 100zł, a potem wykonał zwrot 2024-02-03 za 40zł. Następnie kupił dnia 2024-02-04 za kwotę 150zł i zwrócił w dniu 2024-02-04 za 100zł (przed wysyłką rozmyślił się i zrezygnował z jednego produktu). Oprócz niniejszego klienta są jeszcze wiadomo inni, ale próbujemy ustalić czy wysokości zwrotów jak i zaksięgowane wpłaty są poprawne.
> 
> Mamy teraz 4 operacje płatnicze - dwie wpłaty za zamówienia i dwa zwroty wpłat. O ile jeszcze zamówienia można by było tak sprawdzić, że wziąć to z GET /order/checkout-forms ID płatności i po nim połączyć potem tabelę z listą historii płatniczych (tabela tworzona z użyciem metody payments/payment-operations). W ten sposób tak, miałbym połączenie ID Zamówienia z ID Płatności i byłoby to unikalne połączenie. Co jednak w sytuacji dopłaty, zmiany sposobu płatności? Czy ID Płatności wtedy nie ulega zmianie?

Nie, id płatności nie ulega zmianie. Dochodzi tutaj natomiast identyfikator dopłaty w **surcharges.id**.  Więcej informacji [tutaj](https://developer.allegro.pl/tutorials/jak-obslugiwac-zamowienia-GRaj0qyvwtR).
> 
> Co więcej, skoro to rozwiąże tylko połowę problemów to niniejsze rozwiązanie jest marnym rozwiązaniem. Natomiast sam login nie jest unikatowy, bo kupujący może wykonać kilka zamówień w danym okresie i kilka zwrotów i wtedy jak mam to połączyć z odpowiednimi zamówieniami i zwrotami?

Można tutaj ewentualnie jeszcze połączyć to według daty płatności, ale **occurredAt** w historii operacji płatniczych nie będzie nigdy równa dacie w polu **payment.finishedAt** na liście zamówień. Możesz to jeszcze przetestować na środowisku testowym.
> 
> Jednym słowem w jaki sposób mam uczynić płatność widniejącą na liście operacji płatniczych jako unikatową **dającą się połączyć z zamówieniami i zwrotami**? Bo sama w sobie tak, jest unikatowa bo ID Płatności istnieje. Natomiast jak mam to połączyć z pozostałymi informacjami?

W tym przypadku numer zamówienia byłby tutaj oczywiście najlepszym rozwiązaniem. Zachęcam Cię do przekazania nam sugestii na naszym forum, w dedykowanym miejscu [Github Discussions](https://github.com/allegro/allegro-api/discussions).

**Comment by ricarderobl** [comment]
Dodałem do dyskusji ten wniosek. Ostatnie pytanie w tym wątku w takim razie. Jaki jest orientacyjny czas odniesienia się do propozycji? Niniejsza zmiana nie byłaby nijak inwazyjna, bo dodanie atrybutu do JSON'a jest proste. Sama struktura polecenia też nie uległaby zmianie, jedynie odpowiedź byłaby o dodatkowe pole, co nie wpłynęłoby nijak na tych co już używają tego endpointu.

**Comment by Lukasz-Zurek** [comment]
Dziękuję. Trudno wskazać orientacyjny czas - nie jest on zależny bezpośrednio ode mnie i mojego zespołu. Przekazuję jednak Twój pomysł od razu do właściwego zespołu.
